src/containers/AccountPin.js

```diff
@@ -6,17 +6,17 @@ import { Actions } from 'react-native-router-flux'
 import AccountPin from '../components/AccountPin'
 import { addAccount, setPin } from '../actions/accounts'
 import { signedTx } from '../actions/transactions'
-import { keccak, brainWalletSign } from '../util/native'
+import { brainWalletSign } from '../util/native'
 import { saveAccount } from '../util/db'
+import store from '../util/store'
 
 const mapStateToPropsEnterPin = (state, ownProps) => ({
-  account: state.accounts.selected,
-  extra: state.transactions.pendingTransaction.rlp
+  account: state.accounts.selected
 })
 
-async function signTransaction (dispatch, account, rlp) {
+async function signTransaction (dispatch, account) {
   try {
-    let hash = await keccak(rlp)
+    let hash = store.getState().transactions.pendingTransaction.rlpHash
```

Comment:
Accessing `store` in actions is a little bit smelly to me. I think it's better handled in a middleware. We can also use `redux-thunk` to get this abstraction.